iptables/iptables.go

```diff
@@ -139,6 +139,35 @@ func (ipt *IPTables) Delete(table, chain string, rulespec ...string) error {
 // List rules in specified table/chain
 func (ipt *IPTables) List(table, chain string) ([]string, error) {
 	args := []string{"-t", table, "-S", chain}
+	return ipt.executeList(args)
+}
+
+// ListChains returns a slice containing the name of each chain in the specified table.
+func (ipt *IPTables) ListChains(table string) ([]string, error) {
+	args := []string{"-t", table, "-S"}
+
+	// result contains the output of iptables -t $TABLE -S
+	result, err := ipt.executeList(args)
+	if err != nil {
+		return nil, err
+	}
+
+	// Iterating on all the rules to find the -P (default) and -N (New)
+	// Format is the following:
+	// -P OUTPUT ACCEPT
+	// -N Custom
+	var chains []string
+	for _, val := range result {
+		if strings.HasPrefix(val, "-P") || strings.HasPrefix(val, "-N") {
+			chains = append(chains, strings.Fields(val)[1])
+		} else {
+			break
```

Comment:
Based on my simplistic tests, it seems that -P and -N are always the first lines (And all the subsequent lines have no -P / -N ).
The goal here is to avoid iterating on thousands of rules if you got a chain with a huge amount of rules. 
Let me check if I can confirm that -P and -N are always the first ones.